latex2mmlc/src/parse.rs

```diff
@@ -763,6 +770,10 @@ fn extract_letters<'source>(
         Node::Operator(op, _) | Node::OperatorWithSpacing { op, .. } => {
             buffer.push(op.into());
         }
+        Node::Text(str_ref) => {
+            let string = str_ref.as_str(buffer).to_string();
+            buffer.push_str(string.as_str());
```

Comment:
I need [extend_from_within](https://doc.rust-lang.org/std/string/struct.String.html#method.extend_from_within) here, but that's night-only...

Reply:
I could have a while loop which takes a char and then pushes it to the end.

Reply:
Ah, I can use this trick:

```rust
    let mut s = String::from("Hello, world!");
    
    unsafe {
        let vec = s.as_mut_vec();
        vec.extend_from_within(0..5);
    }
```

Reply:
This produces worse code, alas.